libsolidity/analysis/StaticAnalyzer.cpp

```diff
@@ -22,16 +22,30 @@
  */
 
 #include <libsolidity/analysis/StaticAnalyzer.h>
-
 #include <libsolidity/analysis/ConstantEvaluator.h>
 #include <libsolidity/ast/AST.h>
+#include <libsolidity/ast/ASTUtils.h>
 #include <liblangutil/ErrorReporter.h>
+
+#include <range/v3/view/enumerate.hpp>
+
 #include <memory>
 
 using namespace solidity;
 using namespace solidity::langutil;
 using namespace solidity::frontend;
 
+namespace
+{
+
+Type const* type(Expression const& _expression)
+{
+	solAssert(!!_expression.annotation().type, "Type requested but not present.");
+	return _expression.annotation().type;
```

Comment:
you could change the return value to a `Type const&`  if the expectation is that it exists and it is also asserted for

Reply:
matter of a different PR